### PR TITLE
Update Balerina exercism problems

### PR DESCRIPTION
**Pull request by SasinduDilshara**

### CODE REVIEW

File: exercises/practice/bob/.docs/instructions.md [hunk 1]
```diff
@@ -17,3 +17,11 @@ Bob only ever answers one of five things:
...
 - **"Whatever."**
   This is what he answers to anything else.
+
+## Examples
+
+input - How are you?
+Bob's response - Sure.
+
+input - I am John
+Bob's response - Whatever.
```
Comment (ErikSchierboom) [inline exercises/practice/bob/.docs/instructions.md:27]: The instructions are synced from https://github.com/exercism/problem-specifications/blob/main/exercises/bob/instructions.md. If you think the instructions could be improved, please first discuss this on the forum: https://forum.exercism.org/c/exercism/4

File: exercises/practice/bob/tests/bob_test.bal [hunk 2]
```diff
@@ -7,218 +7,205 @@ function testStatingSomething() {
     test:assertEquals(response, expected);
 }
 
-@test:Config {
-    enable: false
-}
+@test:Config {}
```
Comment (ErikSchierboom) [inline exercises/practice/bob/tests/bob_test.bal:10]: Could you revert these changes? We have a policy to support TDD where students solve tests one at a time.

File: exercises/practice/bob/tests/bob_test.bal [hunk 3]
```diff
@@ -7,218 +7,205 @@ function testStatingSomething() {
...
 function testNonQuestionEndingWithWhitespace() {
     string response = hey("This is a statement ending with whitespace      ");
     string expected = "Whatever.";
     test:assertEquals(response, expected);
 }
+
+@test:Config {}
```
Comment (ErikSchierboom) [inline exercises/practice/bob/tests/bob_test.bal:178]: Tests are synced from a shared source: https://github.com/exercism/problem-specifications/blob/main/exercises/bob/canonical-data.json If you feel like new test cases make sense, they could be added, but just wanted to let you know.

File: exercises/practice/collatz-conjecture/.docs/instructions.md [hunk 4]
```diff
@@ -1,14 +1,14 @@
 # Instructions
 
-The Collatz Conjecture or 3x+1 problem can be summarized as follows:
+The Collatz Conjecture, also known as the 3x+1 problem, can be summarized as follows:
```
Comment (ErikSchierboom) [inline exercises/practice/collatz-conjecture/.docs/instructions.md:3]: Same comment as the bob instructions.

File: exercises/practice/collatz-conjecture/collatz_conjecture.bal [hunk 5]
```diff
@@ -1,3 +1,7 @@
+# Returns the number of steps required to reach 1.
+#
+# + n - number
+# + return - number of steps
```
Comment (ErikSchierboom) [inline exercises/practice/collatz-conjecture/collatz_conjecture.bal:4]: Nice

File: exercises/practice/collatz-conjecture/tests/collatz_conjecture_test.bal [hunk 6]
```diff
@@ -5,30 +5,22 @@ function testZeroStepsForOne() {
     test:assertEquals(collatzSteps(1), 0);
 }
 
-@test:Config {
-    enable: false
-}
+@test:Config {}
```
Comment (ErikSchierboom) [inline exercises/practice/collatz-conjecture/tests/collatz_conjecture_test.bal:8]: Same comment as before

**Review by ErikSchierboom — CHANGES_REQUESTED** [review]

File: exercises/practice/bob/.docs/instructions.md (hunk 1 above)
Comment (SasinduDilshara) [reply exercises/practice/bob/.docs/instructions.md:27]: ok, sure.

Comment (SasinduDilshara) [reply exercises/practice/bob/.docs/instructions.md:27]: Created. https://forum.exercism.org/t/add-example-scenarios-for-the-description-of-the-bob-example-in-ballerina-exercises/5621
Will add a discussions for other scenarios depending on the conclusions of this discussion.

Comment (SasinduDilshara) [reply exercises/practice/bob/.docs/instructions.md]: Removed the changes made for the instructions

File: exercises/practice/bob/tests/bob_test.bal (hunk 2 above)
Comment (SasinduDilshara) [reply exercises/practice/bob/tests/bob_test.bal]: Done

**Review by ThisaruGuruge — APPROVED** [review]

**Comment by ErikSchierboom** [comment]
FYI The reference implementation isn't shown anywhere, so in theory it doesn't have to be idiomatic.
